openforcefield/tests/test_molecule.py

```diff
@@ -303,16 +303,126 @@ def test_to_from_smiles(self, molecule, toolkit):
+                     'rdkit_output': '[H][C]([Cl])=[C]([H])[Cl]',
+                     'isomeric': False, 'explicit_hydrogens': True, 'mapped': False, 'error': None},
+                    {'molecule_input': 'Cl/C=C\Cl', 'openeye_output': None, 'rdkit_output': None,
+                     'isomeric': False, 'explicit_hydrogens': False, 'mapped': True, 'error': AssertionError},
+                    {'molecule_input': 'Cl/C=C\Cl', 'openeye_output': 'C(=CCl)Cl',
+                     'rdkit_output': 'ClC=CCl',
+                     'isomeric': False, 'explicit_hydrogens': False, 'mapped': False, 'error': None},
+                    ]
+
+    @pytest.mark.parametrize('toolkit_class', [OpenEyeToolkitWrapper, RDKitToolkitWrapper])
+    @pytest.mark.parametrize('data', smiles_types)
+    def test_smiles_types(self, data, toolkit_class):
+        """Test that the toolkit is passing the correct args to the toolkit backends across different combinations."""
+
+        if toolkit_class.is_available():
+            toolkit = toolkit_class()
+            mol = Molecule.from_smiles(data['molecule_input'], toolkit_registry=toolkit)
+            isomeric, explicit_hs, mapped = data['isomeric'], data['explicit_hydrogens'], data['mapped']
+            if data['error'] is not None:
+                with pytest.raises(data['error']):
+                    mol.to_smiles(isomeric=isomeric, explicit_hydrogens=explicit_hs,
+                                  mapped=mapped, toolkit_registry=toolkit)
+
+            else:
+                # gather the toolkit dependent output
+                if 'RDKit' in toolkit.__class__.__name__:
+                    output = data['rdkit_output']
+
+                elif 'OpenEye' in toolkit.__class__.__name__:
+                    output = data['openeye_output']
+
+                assert output == mol.to_smiles(isomeric=isomeric, explicit_hydrogens=explicit_hs,
+                                               mapped=mapped, toolkit_registry=toolkit)
+        else:
+            pytest.skip(f'The required toolkit ({toolkit_class.toolkit_name}) is not available.')
+
+    @pytest.mark.parametrize('toolkit_class', [OpenEyeToolkitWrapper, RDKitToolkitWrapper])
+    def test_smiles_chache(self, toolkit_class):
+        """Make sure that the smiles cache is being used correctly."""
+
+        if toolkit_class.is_available():
+            toolkit = toolkit_class()
+            # this uses no toolkit back end so no smiles should be saved
+            mol = create_ethanol()
+
+            # now lets populate the cache with a test result
+            # first we need to make the cache key for the default input
+            isomeric, explicit_hydrogens, mapped = True, True, False
+            cache_key = toolkit.to_smiles.__qualname__ + str(isomeric) + str(explicit_hydrogens) + str(mapped)
+            cache_key += str(mol._properties.get('atom_map', None))
+            mol._cached_smiles = {cache_key: None}
```

Comment:
I don't _love_ that we're reaching down into private attributes here, but I can't think of any other way to run this test, and I think it's good that you thought to test it!